components/cam/src/physics/cam/phys_grid.F90

```diff
@@ -2433,52 +2437,107 @@ subroutine print_cost_p()
+                                      !  speed-up (relative to serial cost)
 
    character(len=*), parameter :: fname = 'atm_chunk_costs.txt'
 
 !-----------------------------------------------------------------------
    if (output_chunk_costs) then
       unitn = getunit()
 
-      ! Calculate normalized measured cost
-      cost_gsum = 0.0_r8
-      cost_lsum = 0.0_r8
+      ! Calculate local measured and estimated chunk costs
+      ! and estimated thread and process costs (for bc_physics and
+      ! ac_physics chunk loops). Estimated thread cost is calculated from
+      ! the chunks assigned to each thread, assuming a static round-robin
+      ! distribution. Estimated process cost is then calculated as the
+      ! maximum over the estimated thread costs.
+      ccest_lsum = 0.0_r8
+      ccost_lsum = 0.0_r8
+      tcest(:)   = 0.0_r8
       do lcid = begchunk, endchunk
-         cost_lsum = cost_lsum + lchunks(lcid)%cost
+         ccest_lsum = ccest_lsum + chunks(lchunks(lcid)%cid)%estcost
+         ccost_lsum = ccost_lsum + lchunks(lcid)%cost
+         tid = mod(lcid-begchunk,nlthreads)
+         tcest(tid) = tcest(tid) + chunks(lchunks(lcid)%cid)%estcost
       enddo
+      pcest = maxval(tcest)
+
 #if ( defined SPMD )
-      call MPI_ALLREDUCE(cost_lsum, cost_gsum, 1, MPI_REAL8, MPI_SUM, &
+      gsums(:) = 0.0_r8
+      lsums(1) = ccest_lsum
+      lsums(2) = ccost_lsum
+      lsums(3) = pcest
+      lsums(4) = phys_pcost
+      call MPI_ALLREDUCE(lsums, gsums, 4, MPI_REAL8, MPI_SUM, &
                          mpicom, ierr)
+      ccest_gsum = gsums(1)
+      ccost_gsum = gsums(2)
+      pcest_sum  = gsums(3)
+      pcost_sum  = gsums(4)
 #else
-      cost_gsum = cost_lsum
+      ccost_gsum = ccost_lsum
+      ccest_gsum = ccest_lsum
+      pcost_sum  = phys_pcost
+      pcest_sum  = pcest
 #endif
-      if (abs(cost_gsum) > 0.000001_r8) then
-         avg_cost = cost_gsum/nchunks
-      else
-         avg_cost = 1.0_r8
+
+      ! Calculate average estimated chunk cost
+      avg_ccest = ccest_gsum/real(nchunks,r8)
+      if (avg_ccest < 0.000001_r8) then
+         avg_ccest = -1.0_r8
       endif
 
-      ! Calculate normalized estimated cost
-      cost_gsum = 0.0_r8
-      do cid = 1,nchunks
-         cost_gsum = cost_gsum + chunks(cid)%estcost
-      enddo
-      if (abs(cost_gsum) > 0.000001_r8) then
-         avg_estcost = cost_gsum/nchunks
-      else
-         avg_estcost = 1.0_r8
+      ! Calculate average measured chunk cost
+      avg_ccost = ccost_gsum/real(nchunks,r8)
+      if (avg_ccost < 0.000001_r8) then
+         avg_ccost = -1.0_r8
+      endif
+
+      ! Calculate average estimated process cost
+      avg_pcest = pcest_sum/real(npes,r8)
+      if (avg_pcest < 0.000001_r8) then
+         avg_pcest = -1.0_r8
+      endif
+
+      ! Calculate average measured process cost
+      avg_pcost = pcost_sum/real(npes,r8)
+      if (avg_pcost < 0.000001_r8) then
```

Comment:
In the spirit of not having "magic numbers" in the code, maybe this threshold value should be assigned to a parameter.